homeassistant/components/seventeentrack/sensor.py

```diff
@@ -210,51 +202,66 @@ def state(self):
     @property
     def unique_id(self):
         """Return a unique, HASS-friendly identifier for this entity."""
-        return ENTITY_ID_TEMPLATE.format(
+        return UNIQUE_ID_TEMPLATE.format(
             self._data.account_id, self._tracking_number)
 
     async def async_update(self):
         """Update the sensor."""
         await self._data.async_update()
 
-        if not self._data.packages:
+        if not self.available:
+            await self._remove()
             return
 
-        try:
-            package = next((
-                p for p in self._data.packages
-                if p.tracking_number == self._tracking_number))
-        except StopIteration:
-            # If the package no longer exists in the data, log a message and
-            # delete this entity:
-            _LOGGER.info(
-                'Deleting entity for stale package: %s', self._tracking_number)
-            reg = await self.hass.helpers.entity_registry.async_get_registry()
-            self.hass.async_create_task(reg.async_remove(self.entity_id))
-            self.hass.async_create_task(self.async_remove())
-            return
+        package = self._data.packages.get(self._tracking_number, None)
 
-        # If the user has elected to not see delivered packages and one gets
+        # If the user has selected to not see delivered packages and one gets
```

Comment:
Unnecessary comment change.

Reply:
well, I was just fixing the typo, do you want me to revert ?

Reply:
It's not a typo.

<img width="626" alt="Screen Shot 2019-07-24 at 1 08 27 PM" src="https://user-images.githubusercontent.com/47216/61821344-2f70a080-ae14-11e9-9e70-dc8bd04878ea.png">

Reply:
sorry :sob:, i'm gonna revert. My English is not perfect

Reply:
No worries! 😄